exercises/linked-list/canonical-data.json

```
@@ -0,0 +1,532 @@
+{
+  "exercise": "linked-list",
+  "comments": [
+    "In order to keep the interfae for the exercise close or equal to the     ",
+    "description.md and also satisfying the canonical-schema.json, the only   ",
+    "properties are pop, push, shift, unshift. Some tracks also implement     ",
+    "delete and count, via a Track-Inserted hints.                            ",
+    "                                                                         ",
+    "It is hard to have interesting tests using the property based approach so",
+    "this canonical data uses the following approach:                         ",
+    "- Each test input is an array of { operation, value?, expected? }        ",
+    "- If operation is push, unshift, delete, then value is given             ",
+    "- If operation is pop, shift, count, the expected value can be given     ",
+    "                                                                         ",
+    "Encoding tests and operations using the same field is necessary to have  ",
+    "tests that don't just initialize the list, and then have one operation.  "
+  ],
+  "cases": [
+    {
+      "uuid": "7f7e3987-b954-41b8-8084-99beca08752c",
+      "description": "pop gets element from the list",
```

Maybe 

```suggestion
      "description": "pop gets element from the singleton list",
```

or 

```suggestion
      "description": "pop gets last element from singleton list",
```

My thinking here being that we might want to specify that pop gets the last element.

---

No, it's purposefully testing you get "a" element. The next test tests that you get the correct element.

---

Okay!

---

I should probably add this as comments?